ably/realtime_conn.go

```diff
@@ -115,7 +115,7 @@ func (c *Connection) dial(proto string, u *url.URL) (conn proto.Conn, err error)
 	query := u.Query()
 	query.Add("heartbeats", "true")
 	u.RawQuery = query.Encode()
-	timeout := c.opts.realtimeRequestTimeout()
+	timeout := c.opts.httpOpenTimeout()
```

Comment:
[From the spec](https://docs.ably.io/client-lib-development-guide/features) (emphasis mine):

> (RTN14c) A new connection attempt will fail if not connected within the default **realtime request timeout**
> (DF1b) realtimeRequestTimeout – default 10s. When a realtime client library is **establishing a connection with Ably**, or sending a HEARTBEAT, CONNECT, ATTACH, DETACH or CLOSE ProtocolMessage to Ably, this is the amount of time that the client library will wait before considering that request as failed and triggering a suitable failure condition

But then:

> (TO3l3) httpOpenTimeout integer – default 4,000 (4s). Timeout for opening the connection, available in the client library if supported by the transport

This is a bit ambiguous but the `http` prefix there suggests to me that `httpOpenTimeout` is meant for connections over which HTTP requests are performed.

Given this change isn't critical for the main purpose of this pull request, maybe we should leave it out?

Reply:
> This is a bit ambiguous but the `http` prefix there suggests to me that `httpOpenTimeout` is meant for connections over which HTTP requests are performed.

Doing a bit of blame archeology, this [indeed seems to be the case](https://github.com/ably/docs/pull/51#discussion_r41497849).

Reply:
> This is a bit ambiguous but the `http` prefix there suggests to me that `httpOpenTimeout` is meant for connections over which HTTP requests are performed.

I agree, I had similar concerns there: https://github.com/ably/ably-boomer/pull/44#issuecomment-839981801

> Given this change isn't critical for the main purpose of this pull request, maybe we should leave it out?

The issue then is that we don't have a way to specify a different connection and request timeout for Realtime. We could use a custom `Dial` to override the timeout value, but the default websocket `Dial` is internal so we cannot use it in ably-boomer. That would mean either copy and pasting that default websocket dial or making it non-internal.

Reply:
Actually what you discussed in https://github.com/ably/ably-boomer/pull/44 sounds to me like a more sensible interpretation of the spec than what I said above. `realtimeRequestTimeout` is for "establishing a connection with Ably", ie. an _Ably realtime_ connection. So it's the timeout for receiving a CONNECTED ProtocolMessage response.

`Dial` is about establishing the _transport_ WebSocket connection on which the Ably realtime connection will be based, so, while there isn't a specific timeout defined for this, `httpOpenTimeout` is close enough.

Reply:
I meant to say that I agree with this change now :) But I don't think the test for RTN14c should be changed. RTN14c is about `realtimeRequestTimeout`.